packages/block-library/src/dialog-element/index.php

```diff
@@ -0,0 +1,245 @@
+			$position_styles = 'margin-right: 1em;';
+			break;
+		case 'bottom left':
+			$position_styles = 'margin-bottom: 1em; margin-left: 1em;';
+			break;
+		case 'bottom center':
+			$position_styles = 'margin-bottom: 1em;';
+			break;
+		case 'bottom right':
+			$position_styles = 'margin-bottom: 1em; margin-right: 1em;';
+			break;
+	}
+
+	return $position_styles;
+}
+
+/**
+ * Render the 'core/dialog-element' block.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content Block content.
+ * @param WP_Block $block Block instance.
+ *
+ * @return string Rendered block HTML.
+ */
+function render_block_core_dialog_element( array $attributes, string $content, WP_Block $block ): string {
+	if ( empty( $content ) ) {
+		return '';
+	}
+
+	$suffix = wp_scripts_get_suffix();
+	if ( defined( 'IS_GUTENBERG_PLUGIN' ) && IS_GUTENBERG_PLUGIN ) {
+		$module_url = gutenberg_url( '/build-module/block-library/dialog-element/view.min.js' );
+	}
+
+	wp_register_script_module(
+		'@wordpress/block-library/dialog-element',
+		isset( $module_url ) ? $module_url : includes_url( "blocks/dialog-element/view{$suffix}.js" ),
+		array( '@wordpress/interactivity' ),
+		defined( 'GUTENBERG_VERSION' ) ? GUTENBERG_VERSION : get_bloginfo( 'version' )
+	);
+
+	wp_enqueue_script_module( '@wordpress/block-library/dialog-element' );
+
+	$context_id = isset( $block->context['core/dialog-id'] ) ? $block->context['core/dialog-id'] : null;
+	if ( ! $context_id ) {
+		return '';
+	}
+	$is_open                 = get_query_var( 'dialogId', false ) === $context_id;
+	$dialog_size             = isset( $attributes['dialogSize'] ) ? $attributes['dialogSize'] : 'small';
+	$animation               = isset( $attributes['animation'] ) ? $attributes['animation'] : 'fade';
+	$auto_activate_on_render = array_key_exists( 'autoActivateOnRender', $attributes ) ? $attributes['autoActivateOnRender'] : false;
+	$default_is_open         = $auto_activate_on_render;
+	$auto_activation_timer   = array_key_exists( 'autoActivationTimer', $attributes ) ? $attributes['autoActivationTimer'] : -1;
+	$auto_activation_timer   = $default_is_open ? 0 : $auto_activation_timer;
+	$enable_deep_link        = array_key_exists( 'enableDeepLink', $attributes ) ? $attributes['enableDeepLink'] : false;
+
+	// By using state any 3rd party can interact as easy as `store('core/dialog').state.dialogs.[blockId].isOpen = true;` which would open the dialog given the blockId.
+	$dialog_state           = wp_interactivity_state(
+		'core/dialog',
+		array()
+	);
+	$dialogs                = $dialog_state['dialogs'] ?? array();
+	$dialogs[ $context_id ] = array(
+		'id'                      => $context_id,
+		'activationTimerDuration' => (int) $auto_activation_timer,
+		'isOpen'                  => $is_open,
+		'enableDeepLink'          => $enable_deep_link,
+		'isClosing'               => false,
+	);
+	wp_interactivity_state(
+		'core/dialog',
+		array(
+			'dialogs' => $dialogs,
+		)
+	);
```

Comment:
You don't need to do this, associative arrays are not replaced, they are merged.

```suggestion
wp_interactivity_state( 'core/dialog', array( 
	'dialogs' => array(
		$context_id => array(
			'id'                      => $context_id,
			'activationTimerDuration' => (int) $auto_activation_timer,
			'isOpen'                  => $is_open,
			'enableDeepLink'          => $enable_deep_link,
			'isClosing'               => false,
		)
	)
) );
```

Reply:
Oh wow, well I have a ton of block clean up to do then. Much cleaner.